### PR TITLE
`Window.showDirectoryPicker()`: Add `mode`

### PR DESCRIPTION
**Pull request by tomayac**
#### Summary

Add `mode` to directory picker.

#### Motivation

Document API improvement.

#### Supporting details

https://bugs.chromium.org/p/chromium/issues/detail?id=1115632

#### Metadata

- [ ] Adds a new document
- [X] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/api/window/showdirectorypicker/index.md [hunk 1]
```diff
@@ -25,7 +25,13 @@ var FileSystemDirectoryHandle = window.showDirectoryPicker();
 
 ### Parameters
 
-None.
+- `options` {{optional_inline}}
+
+  - : An object containing options, which are as follows:
+
+    - `mode`
+      - : A string that defaults to `read` for read-only access or `readwrite` for read
```
Comment (tomayac) [inline files/en-us/web/api/window/showdirectorypicker/index.md:33]:
```suggestion
      - : A string that defaults to `"read"` for read-only access or `"readwrite"` for read
```

**Comment by teoli2003** [comment]
I agree with should add it (as well as the missing field `id` and `startIn` options.

Before actually reviewing the change here, I have a question, is there any reason the `mode` option has not been added to the [proposed spec](https://wicg.github.io/file-system-access/#local-filesystem)? I would like to know if we should add `{{non-standard_Inline}} next to it or not (and without it into the spec, it would be tricky to reach interoperability).

We will also need an entry in bcd.

**Comment by tomayac** [comment]
> I agree with should add it (as well as the missing field `id` and `startIn` options.

Add the two missing ones.

> Before actually reviewing the change here, I have a question, is there any reason the `mode` option has not been added to the [proposed spec](https://wicg.github.io/file-system-access/#local-filesystem)? I would like to know if we should add `{{non-standard_Inline}} next to it or not (and without it into the spec, it would be tricky to reach interoperability).

This is the PR in question: https://github.com/WICG/file-system-access/pull/300.

> We will also need an entry in bcd.

Right. I need to find out how to do this.

**Comment by teoli2003** [comment]
For bcd, you need to add a subfeature under https://github.com/mdn/browser-compat-data/blob/main/api/Window.json#L5767

**Review by teoli2003 — CHANGES_REQUESTED** [review]
The last bit here:
On line 11 add an extra tag ` - Experimental`.
and on line 14, add `{{SeeCompatTable}}`

This will mark the feature as experimental (as only one engine supports it)

Even if the spec PR is not merged yet, given only an administrative burden prevents it, we will skip the _Non-standard_ inline flag for `mode`. (No need to have another PR here in a coupe of weeks).

**Review by teoli2003 — APPROVED** [review]
Thank you! Let's merge this!

**Pull request merged**